### PR TITLE
feat: invoke calculate fees v2

### PR DESCRIPTION
**Pull request by pietro-tota**
Depends on https://github.com/pagopa/pagopa-infra/pull/1973
Depends on https://github.com/pagopa/pagopa-ecommerce-payment-methods-service/pull/186

#### List of Changes
Add implementation for use calculate fees for carts payments

#### Motivation and Context
Add implementation for use calculate fees for carts payments

#### How Has This Been Tested?
Local tests with pagopa-checkout-be-mock

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
...
-    "generate:payment-ecommerce-v2": "rimraf generated/definitions/payment-ecommerce-v2 && shx mkdir -p generated/definitions/payment-ecommerce-v2 && gen-api-models --api-spec https://raw.githubusercontent.com/pagopa/pagopa-infra/main/src/domains/ecommerce-app/api/ecommerce-checkout/v2/_openapi.json.tpl --no-strict --out-dir ./generated/definitions/payment-ecommerce-v2 --request-types --response-decoders --client",
+    "generate:payment-ecommerce-v2": "rimraf generated/definitions/payment-ecommerce-v2 && shx mkdir -p generated/definitions/payment-ecommerce-v2 && gen-api-models --api-spec https://raw.githubusercontent.com/pagopa/pagopa-infra/CHK-2798-openapi-multi-fee/src/domains/ecommerce-app/api/ecommerce-checkout/v2/_openapi.json.tpl --no-strict --out-dir ./generated/definitions/payment-ecommerce-v2 --request-types --response-decoders --client",
```
Comment (pietro-tota) [inline package.json:9]:
```suggestion
    "generate:payment-ecommerce-v2": "rimraf generated/definitions/payment-ecommerce-v2 && shx mkdir -p generated/definitions/payment-ecommerce-v2 && gen-api-models --api-spec https://raw.githubusercontent.com/pagopa/pagopa-infra/main/src/domains/ecommerce-app/api/ecommerce-checkout/v2/_openapi.json.tpl --no-strict --out-dir ./generated/definitions/payment-ecommerce-v2 --request-types --response-decoders --client",
```
TODO: to be replaced after dependent infra pr merge

File: src/utils/api/client.ts [hunk 2]
```diff
@@ -54,3 +54,18 @@ export const apiPaymentEcommerceClientWithRetry = createEcommerceClient({
     async (r: Response): Promise<boolean> => r.status > 499
   ),
 });
+
+/**
+ * Api client for ecommerce API calculate fee with retry execution
+ */
+export const apiPaymentEcommerceClientWithRetryV2 = createEcommerceClientV2({
```
Comment (vito80ba) [inline src/utils/api/client.ts:61]: Why create a brand new client with the second version of EccommerceClient and use it for "fees"?
Is it correct to assume that only Fees will use it and not in the other points of the webapp? Cause, if this assumption is not true, maybe it's better to modify the existing one `apiPaymentEcommerceClientWithRetry`

Comment (pietro-tota) [reply src/utils/api/client.ts:61]: `apiPaymentEcommerceClientWithRetry` is actually used for the following api calls:
- requestTransactionAuthorization
- createSession

eCommerce side only modified api will be included in the next version (v2 by the moment) and this is the case for the calculateFees one (for handle calculate fees with multiple payment notices).
So by the moment it's mandatory to leave both clients, one per v1 api's, one for v2 ones

**Review by vito80ba — APPROVED** [review]

**Review by petretiandrea — APPROVED** [review]
🚀

File: .devops/pagopa-code-review-pipelines.yml [hunk 3]
```diff
@@ -34,7 +34,7 @@ resources:
     - repository: checkout-be-mock
       type: github
       name: pagopa/pagopa-checkout-be-mock
-      ref: master
+      ref: CHK-2800-calculate-fees-multi
```
Comment (pietro-tota) [inline .devops/pagopa-code-review-pipelines.yml:37]:
```suggestion
      ref: master
```
TODO: to be restored after dependent PR merge

**Pull request merged**